### ISSUE TITLE
[v4] custom function returning a single custom data structure lead to "type does not exist"

### ISSUE THREAD

**Issue by EyMaddis** [postgraphql/postgraphql#555]
I use a custom function called `create_session` defined like this:
```sql
CREATE OR REPLACE FUNCTION public.create_session(
  client_id functionSchema.GUID,
  sub functionSchema.GUID,
  exp timestamp,
  iat timestamp,
  iss varchar(255),
  role varchar(255),
  aud text,
  ip inet,
  ip_location text
)
RETURNS functionSchema.GUID
VOLATILE
SECURITY DEFINER
AS $function$
...
$function$;
```

In v4 this throws the internal (Postgres) error `type "functionSchema.guid[]" does not exist`.

Error logs:
```
postgraphql:request GraphQL query request has begun. +19ms
postgraphql:request GraphQL query is parsed. +0ms
postgraphql:request GraphQL query is validated. +1ms
postgraphql:postgres begin +0ms
postgraphql:postgres select set_config($1, $2, true), set_config($3, $4, true), set_config($5, $6, true), set_config($7, $8, true), set_config($9, $10, true), set_config($11, $12, true), set_config($13, $14, true) +1ms
postgraphql:postgres 
postgraphql:postgres       with __local_0__ as (
postgraphql:postgres         select __local_1__.__local_1__ as __local_0__ from "public"."create_session"(NULL, $1, $2, $3, $4, $5, $6, NULL, NULL) __local_1__
postgraphql:postgres       )
postgraphql:postgres       select __local_0__.__local_0__::"functionSchema"."guid" from __local_0__ +2ms
postgraphql:postgres 
postgraphql:postgres       with __local_0__ as (
postgraphql:postgres         select str::"functionSchema"."guid" as __local_0__
postgraphql:postgres         from unnest(($1)::"functionSchema"."guid"[]) str
postgraphql:postgres       )
postgraphql:postgres 
postgraphql:postgres       select to_json(__local_0__.__local_0__) as "value"
postgraphql:postgres       from __local_0__ as __local_0__
postgraphql:postgres 
postgraphql:postgres       where (TRUE) and (TRUE)
postgraphql:postgres 
postgraphql:postgres 
postgraphql:postgres 
postgraphql:postgres      +2ms
postgraphql:postgres:error error: type "functionSchema.guid[]" does not exist
postgraphql:postgres:error     at Connection.parseE (/srv/webservice/node_modules/pg/lib/connection.js:554:11)
postgraphql:postgres:error     at Connection.parseMessage (/srv/webservice/node_modules/pg/lib/connection.js:381:17)
postgraphql:postgres:error     at Socket.<anonymous> (/srv/webservice/node_modules/pg/lib/connection.js:117:22)
postgraphql:postgres:error     at emitOne (events.js:115:13)
postgraphql:postgres:error     at Socket.emit (events.js:210:7)
postgraphql:postgres:error     at addChunk (_stream_readable.js:266:12)
postgraphql:postgres:error     at readableAddChunk (_stream_readable.js:253:11)
postgraphql:postgres:error     at Socket.Readable.push (_stream_readable.js:211:10)
postgraphql:postgres:error     at TCP.onread (net.js:585:20) +1ms
postgraphql:postgres commit +0ms
postgraphql:request GraphQL query has been executed. +1ms
postgraphql:request GraphQL query request finished. +0ms
error(s) as <userrole> in 8.74ms :: mutation createSession($input: CreateSessionInput!) { createSession(input: $input) { guid } }
```

The same query in v3 logs the following and works as expected:

```
postgraphql:request GraphQL query request has begun. +19ms
postgraphql:request GraphQL query is parsed. +0ms
postgraphql:request GraphQL query is validated. +1ms
postgraphql:postgres begin +0ms
postgraphql:postgres select set_config($1, $2, true), set_config($3, $4, true), set_config($5, $6, true), set_config($7, $8, true), set_config($9, $10, true), set_config($11, $12, true), set_config($13, $14, true) +1ms
postgraphql:postgres select to_json("public"."create_session"(null, $1, $2, $3, $4, $5, $6, null, null)) as value +2ms
postgraphql:postgres commit +5ms
postgraphql:request GraphQL query has been executed. +3ms
postgraphql:request GraphQL query request finished. +0ms
rror(s) as <userRole> in 12.15ms :: mutation createSession($input: CreateSessionInput!) { createSession(input: $input) { guid } }
```

**Comment by benjie** [comment]
Interesting; it seems your schema doesn't have a default array type for the `GUID` type. I'm wondering if this is because the type itself is, effectively, an array type?

https://www.postgresql.org/docs/9.6/static/sql-createtype.html#AEN84084

Could you share with me the GUID type itself - did it come from a PostgreSQL extension or is it of your own creation? The `CREATE TYPE` statement required for it (e.g. from a pg_dump) would be useful, as would:

```sql
select * from pg_type where typname like '%guid%';
```

**Comment by benjie** [comment]
(Also: what PostgreSQL version?)

**Comment by EyMaddis** [comment]
Sure.
It is a varchar domain type in PostgreSQL 9.6.1: 

```CREATE DOMAIN smpl.GUID
 AS character varying(15)
 DEFAULT '000000000000000'::character varying
 CONSTRAINT guid_conformity CHECK (VALUE::text ~ '^[a-zA-Z0-9]{15}$'::text);
```

**Comment by benjie** [comment]
Should be resolved in `postgraphql@4.0.0-alpha2.12` (`postgraphql@next`)
